mir_eval/display.py

```diff
@@ -0,0 +1,672 @@
+    style.update(kwargs)
+
+    if midi:
+        idx = frequencies > 0
+        frequencies[idx] = hz_to_midi(frequencies[idx])
+
+        # Tick at integer midi notes
+        ax.yaxis.set_minor_locator(MultipleLocator(1))
+
+    for idx in v_slices:
+        ax.plot(times[idx], frequencies[idx], **style)
+        style.pop('label', None)
+
+    # Plot the unvoiced portions
+    if unvoiced:
+        style['alpha'] = style.get('alpha', 1.0) * 0.5
+        for idx in u_slices:
+            ax.plot(times[idx], frequencies[idx], **style)
+
+    return ax
+
+
+def multipitch(times, frequencies, midi=False, unvoiced=False, ax=None,
+               **kwargs):
+    '''Visualize multiple f0 measurements
+
+    Parameters
+    ----------
+    times : np.ndarray, shape=(n,)
+        Sample times of frequencies
+
+    frequencies : list of np.ndarray
+        frequencies (in Hz) of the pitch measurements.
+        Voicing is indicated by sign (positive for voiced,
+        non-positive for non-voiced).
+
+        `times` and `pitches` should be in the format produced by
+        :func:`mir_eval.io.load_ragged_time_series`
+
+    midi : bool
+        If `True`, plot on a MIDI-numbered vertical axis.
+        Otherwise, plot on a linear frequency axis.
+
+    unvoiced : bool
+        If `True`, unvoiced pitches are plotted and indicated
+        by transparency.
+
+        Otherwise, unvoiced pitches are omitted from the display.
+
+    ax : matplotlib.pyplot.axes
+        An axis handle on which to draw the intervals.
+        If none is provided, a new set of axes is created.
+
+    kwargs :
+        Additional keyword arguments to `plt.scatter`.
+
+    Returns
+    -------
+    ax
+        Handle to the plotting axes
+    '''
+
+    if ax is None:
+        ax = plt.gca()
+
+    # Set up a style for the plot
+    style_voiced = dict()
+    style_voiced.update(next(ax._get_lines.prop_cycler))
+    style_voiced.update(kwargs)
+
+    style_unvoiced = style_voiced.copy()
+    style_unvoiced.pop('label', None)
+    style_unvoiced['alpha'] = style_unvoiced.get('alpha', 1.0) * 0.5
+
+    # We'll collect all times and frequencies first, then plot them
+    voiced_times = []
+    voiced_freqs = []
+
+    unvoiced_times = []
+    unvoiced_freqs = []
+
+    for t, freqs in zip(times, frequencies):
+        if not len(freqs):
+            continue
+
+        freqs, voicings = freq_to_voicing(np.asarray(freqs, dtype=np.float))
+
+        # Discard all 0-frequency measurements
+        idx = freqs > 0
+        freqs = freqs[idx]
+        voicings = voicings[idx]
+
+        if midi:
+            freqs = hz_to_midi(freqs)
+
+        n_voiced = sum(voicings)
+        voiced_times.extend([t] * n_voiced)
+        voiced_freqs.extend(freqs[voicings])
+        unvoiced_times.extend([t] * (len(freqs) - n_voiced))
+        unvoiced_freqs.extend(freqs[~voicings])
+
+    # Plot the voiced frequencies
+    ax.scatter(voiced_times, voiced_freqs, **style_voiced)
```

Comment:
This reminded me that I challenged @justinsalamon to do clustering here: https://github.com/craffel/mir_eval/pull/196#issuecomment-223031002 "If you're feeling inspired @justinsalamon maybe you can implement [TF lustering] as a fork of @bmcfee's progress and we can decide from there?"  Just a final ping on that